volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -213,65 +213,63 @@ def get_elf_table_name(self):
         )
         return elf_table_name
 
-    def get_symbols(self):
+    def get_symbols(
+        self,
+    ) -> Iterable[Tuple[int, interfaces.objects.ObjectInterface]]:
         """Get symbols of the module
 
         Yields:
-                A symbol object
+            A tuple containing the ELF symbol index and the corresponding ELF symbol object
         """
 
-        if not hasattr(self, "_elf_table_name"):
-            self._elf_table_name = self.get_elf_table_name()
-        if symbols.symbol_table_is_64bit(self._context, self.get_symbol_table_name()):
-            prefix = "Elf64_"
-        else:
-            prefix = "Elf32_"
-        syms = self._context.object(
-            self.get_symbol_table_name() + constants.BANG + "array",
+        if not self.section_strtab or self.num_symtab < 1:
+            return None
+
+        elf_table_name = self.get_elf_table_name()
+        symbol_table_name = self.get_symbol_table_name()
+
+        is_64bit = symbols.symbol_table_is_64bit(self._context, symbol_table_name)
+        sym_name = "Elf64_Sym" if is_64bit else "Elf32_Sym"
+        sym_type = self._context.symbol_space.get_type(
+            elf_table_name + constants.BANG + sym_name
+        )
+        elf_syms = self._context.object(
+            symbol_table_name + constants.BANG + "array",
             layer_name=self.vol.layer_name,
             offset=self.section_symtab,
-            subtype=self._context.symbol_space.get_type(
-                self._elf_table_name + constants.BANG + prefix + "Sym"
-            ),
-            count=self.num_symtab + 1,
+            subtype=sym_type,
+            count=self.num_symtab,
```

Comment:
Did the previous version have a bug (and set the count too high) or was missing the `+ 1` a typo?  Happy with either outcome, just checking which...

Reply:
Yeap, bug. That's the first bullet:
> Fixed an issue causing the generation of an invalid, extra symbol.